pywavefront/material.py

```diff
@@ -166,24 +183,57 @@ def parse_Ns(self):
 
     @auto_consume
     def parse_d(self):
-        self.this_material.set_alpha(self.values[1])
+        """Transparency"""
+        self.this_material.set_alpha(float(self.values[1]))
+
+    @auto_consume
+    def parse_Tr(self):
+        """Transparency (alternative)"""
+        self.this_material.set_alpha(1.0 - float(self.values[1]))
 
     @auto_consume
     def parse_map_Kd(self):
+        """Diffuse map"""
         Kd = os.path.join(self.dir, " ".join(self.values[1:]))
         self.this_material.set_texture(Kd)
 
     @auto_consume
-    def parse_Ni(self):
-        # unimplemented
-        pass
+    def parse_map_Ka(self):
+        """Ambient map"""
+        Kd = os.path.join(self.dir, " ".join(self.values[1:]))
+        self.this_material.set_texture_ambient(Kd)
 
     @auto_consume
-    def parse_Tr(self):
-        # unimplemented
+    def parse_map_Ks(self):
+        """Specular color map"""
+        Kd = os.path.join(self.dir, " ".join(self.values[1:]))
+        self.this_material.set_texture_specular_color(Kd)
+
+    @auto_consume
+    def parse_map_Ns(self):
+        """Specular color map"""
+        Kd = os.path.join(self.dir, " ".join(self.values[1:]))
+        self.this_material.set_texture_specular_highlight(Kd)
+
+    @auto_consume
+    def parse_map_d(self):
+        """Aplha map"""
+        Kd = os.path.join(self.dir, " ".join(self.values[1:]))
+        self.this_material.set_texture_alpha(Kd)
+
+    @auto_consume
+    def parse_map_bump(self):
+        """Bump map"""
+        Kd = os.path.join(self.dir, " ".join(self.values[1:]))
+        self.this_material.set_texture_bump(Kd)
+
+    def parse_bump(self):
+        self.parse_map_bump()
+
+    @auto_consume
+    def parse_Ni(self):
         pass
 
     @auto_consume
     def parse_illum(self):
-        # unimplemented
```

Comment:
In accordance with the other changes here, should `parse_illum` be removed entirely?

Reply:
I'll just implement them instead since they have been around.